src/EditorFeatures/Test/CodeFixes/ErrorCases/CodeFixExceptionInRegisterMethodAsync.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Composition;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.CodeFixes.ErrorCases
+{
+    public class ExceptionInRegisterMethodAsync : CodeFixProvider
```

Comment:
Please test cases where exception is thrown from within extendable APIs in CodeAction (e.g. ComputeOperationsAsync(), ComputePreviewOperationsAsync(), Title, .EquivalenceKey, GetChangedSolutionAsync(), GetChangedDocumentAsync(), PostProcessChangesAsync() etc.)

Also test this in IDE to make sure we don't crash when there is exception being thrown from these paths (Looks like .EquivalenceKey usages in SuggestedAction.cs and CodeFixSuggestedAction.cs are not being surrounded with extensionManager.PerformFunction() - we should probably introduce a property for EquivalenceKey on SuggestedAction that accesses CodeAction.EquivalenceKey under an extensionManager.PerformFunction() call and caches the value so that we only do this once for repeated accesses).